e2e/testcases/helm_sync_test.go

```diff
@@ -289,6 +470,86 @@ func TestHelmNamespaceRepo(t *testing.T) {
 	}
 }
 
+// TestHelmNamespaceRepoWithConfigMap verifies RepoSync can pick up values and updates from ConfigMap references.
+// This test will work only with following pre-requisites:
+// Google service account `e2e-test-ar-reader@${GCP_PROJECT}.iam.gserviceaccount.com` is created with `roles/artifactregistry.reader` for accessing images in Artifact Registry.
+func TestHelmConfigMapNamespaceRepo(t *testing.T) {
+	repoSyncNN := nomostest.RepoSyncNN(testNs, configsync.RepoSyncName)
+	nt := nomostest.New(t, nomostesting.SyncSource, ntopts.RequireGKE(t),
+		ntopts.RepoSyncPermissions(policy.AllAdmin()), // NS reconciler manages a bunch of resources.
```

Comment:
Can this be more restrictive than `AllAdmin`? What resources are being managed here? Using overkill permissions here can mask other issues (e.g. the base ClusterRole has insufficient permissions)

Reply:
It needs to manage the resources defined in the [test chart](https://github.com/GoogleContainerTools/kpt-config-sync/tree/main/e2e/testdata/helm-charts/ns-chart/templates). I didn't write the original chart but it seems to me that the HPA doesn't need to be there necessarily. I don't see this chart used anywhere else besides the helm-sync tests, so I took out the HPA. 

It does also need `CoreAdmin`; otherwise it gets this error: 
```polling for status failed: failed to list /v1, Kind=Pod: pods is forbidden: User "system:serviceaccount:config-management-system:ns-reconciler-test-ns" cannot list resource "pods" in API group "" in the namespace "test-ns"
```

So I changed it from `AllAdmin` to just `CoreAdmin`, `AppsAdmin`, and `RepoSyncAdmin`. 

I did some manual testing (applied a RepoSync directly to my cluster) to check that the base ClusterRole has sufficient permissions and it is working fine - helm-sync got the ConfigMap values and successfully rendered the chart.